tests/util_test/util.hpp

```diff
@@ -310,4 +336,27 @@ inline auto make_addr(uint8_t i) {
   return ret;
 }
 
+using expected_balances_map_t = std::map<addr_t, u256>;
+inline void wait_for_balances(const vector<std::shared_ptr<FullNode>>& nodes, const expected_balances_map_t& balances,
+                              wait_opts to_wait = {10s, 500ms}) {
+  TransactionClient trx_client(nodes[0]);
+  auto sendDummyTransaction = [&]() {
+    trx_client.coinTransfer(KeyPair::create().address(), 0, KeyPair::create(), false);
+  };
+  wait(to_wait, [&](auto& ctx) {
+    for (const auto& node : nodes) {
+      for (const auto& b : balances) {
+        if (node->getFinalChain()->getBalance(b.first).first != b.second) {
+          sendDummyTransaction();
+          ctx.fail();
```

Comment:
So how does ctx.fail() replace EXPECT_EQ ?

Reply:
If we are not getting equal balances during the set time above we will get an error here

Reply:
yeah so my question is. That error that we get behaves in similar manner to what we get when using condition like `EXPECT_EQ`, which prints also values that are compared etc... ?

Reply:
Didn't thought about that. I will replace it there and fail_if below to get more explicit error logs